packages/headless/src/controllers/commerce/facets/core/headless-core-commerce-facet.ts

```diff
@@ -271,17 +277,17 @@ export function buildCoreCommerceFacet<
     get state() {
       const response = getResponse();
 
-      const values = response.values as ValueResponse[];
+      const values = (response?.values ?? []) as ValueResponse[];
       const hasActiveValues = values.some(
         (facetValue) => facetValue.state !== 'idle'
       );
-      const canShowMoreValues = response ? response.moreValuesAvailable : false;
+      const canShowMoreValues = response?.moreValuesAvailable ?? false;
 
       return {
         facetId,
-        type: response.type,
-        field: response.field,
-        displayName: response.displayName,
+        type: response?.type ?? 'regular',
```

Comment:
It feels a bit awkward to return a facet state at all when there's no facet response.

Maybe get state() should just return an empty object in such a case... What do you think?

Reply:
Returning an empty object would force us to change the state's type. Since this is kind of a hack-ish fix, I'd rather keep things as-is.

When we introduce the concept of routes, we should be able to revert some of the changes in this PR. Facets, sorting and pagination will be set on a PLP's (or search's) state, and not on a global state on the engine, which means we won't need to handle this interference we're fixing here.